docs/examples.rst

```diff
@@ -0,0 +1,310 @@
+one:
+
+.. code:: python
+
+    import os
+    import yogadl
+    import yogadl.storage
+
+    # Create a yogadl.Storage object backed by the local filesystem.
+    storage_path = "/tmp/yogadl_cache"
+    os.makedirs(storage_path, exist_ok=True)
+    lfs_config = yogadl.storage.LFSConfigurations(storage_path)
+    storage = yogadl.storage.LFSStorage(lfs_config)
+
+
+Storing a dataset
+^^^^^^^^^^^^^^^^^
+
+Let's create a silly 10-record dataset and store it in the ``yogadl.Storage``.
+This is done via ``storage.submit()``.  During ``storage.submit()``, the entire
+dataset will be read and written to a the storage backend (in this case, to a
+file).
+
+.. code:: python
+
+    import tensorflow as tf
+
+    # Create a dataset we can store.
+    records = tf.data.Dataset.range(10)
+
+    # Store this dataset as "range" version "1.0"
+    storage.submit(records, "range", "1.0")
+
+
+Fetching a previously stored dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Later (probably in a different process), you can fetch a
+:class:`yogadl.DataRef` representing the dataset via ``storage.fetch()``.
+
+A ``DataRef`` is just a reference to a dataset.  In this case, the dataset will
+be stored in a file on your computer, but a ``DataRef`` could just as easily
+refer to a dataset on some remote machine; the interface would be the same.
+
+To actually access the dataset, you need to first call ``dataref.stream()``,
+which will return a :class:`yogadl.Stream` object.  Then you can convert the
+``Stream`` object to a framework-native data loader format (currently only
+``tf.data.Dataset`` is supported).
+
+.. code:: python
+
+    import yogadl.tensorflow
+
+    # Get the DataRef.
+    dataref = storage.fetch("range", "1.0")
+
+    # Tell the DataRef how to stream the dataset.
+    stream = dataref.stream(start_offset=5, shuffle=True, shuffle_seed=777)
+
+    # Interpret the stream as a tensorflow dataset
+    records = yogadl.tensorflow.make_tf_dataset(stream)
+
+    # It's a real tf.data.Dataset; you can use normal tf.data operations on it.
+    batches = records.repeat(3).batch(5)
+
+    # (this part requires TensorFlow >= 2.0)
+    for batch in batches:
+        print(batch)
+
+This should print:
+
+.. code::
+
+    tf.Tensor([5 1 9 6 7], shape=(5,), dtype=int64)
+    tf.Tensor([1 7 3 9 8], shape=(5,), dtype=int64)
+    tf.Tensor([2 6 0 4 5], shape=(5,), dtype=int64)
+    tf.Tensor([9 5 3 0 8], shape=(5,), dtype=int64)
+    tf.Tensor([6 7 4 1 2], shape=(5,), dtype=int64)
+
+Notice that:
+
+ - The start_offset is only applied to the first epoch, so in this example
+   .repeat(3) gave us 2.5 epochs of data since we skipped the first epoch.
+ - The shuffle is a true shuffle. The shuffled stream samples from the whole
+   dataset without any concept of a "buffer".
+ - The shuffle is reproducible because we chose a shuffle seed.
+ - Each epoch is reshuffled.  Also the reshuffle is reproducible; if you set a
```

Comment:
I found everything after `Each epoch is reshuffled.` hard to follow.

Reply:
I'll just take it out.  It's probably a little too much detail.